### PR TITLE
Fixes for jump table ##analysis

### LINKED ISSUE #304: Switch statement edges duplicated and attached in wrong place
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | macOS/x86_64 on arm64 (with rosetta)
| File format of the file you reverse (mandatory)      | machO
| Architecture/bits of the file (mandatory)            | x86 64
| rizin -v full output, **not truncated** (mandatory)  | rizin 0.1.0 0 @ darwin-x86-64 git.0.1.0 commit: unknown build: 2021-01-04__22:56:50

Not sure why commit hash wasn't properly included in `rizin -v`. Should be 4ac005d4894ac5f13b007784aad4d6147ebf1296

### Expected behavior

Switch statement edges are attached to the basic block which ends with computed jump

### Actual behavior

Switch statement edges are added twice and to two basic blocks which precede the block with indirect jump.

### Steps to reproduce the behavior 
* Open /bin/ls on macOS
* aaa
* view CFG for main function.

### Additional Logs, screenshots, source-code,  configuration dump, ...

```asm
0x100004153      mov     byte [0x100008504], 1 ; switch statement edges attached here 
0x10000415a      mov     r12d, 0x10 ; and here
0x100004160      lea     r13, str.1_ABCFGHLOPRSTUWabcdefghiklmnopqrstuvwx ; 0x100007a9c
0x100004167      lea     rbx, [0x1000048e8]
0x10000416e      mov     edi, r15d ; int argc
0x100004171      mov     rsi, r14  ; const char * *argv
0x100004174      mov     rdx, r13  ; const char *optstring
0x100004177      call    getopt    ; sym.imp.getopt ; int getopt(int argc, const char * *argv, const char *optstring)
0x10000417c      lea     ecx, [rax - 0x25]
0x10000417f      cmp     ecx, 0x53
0x100004182      ja      case.default.0x10000418b
0x100004184      movsxd  rax, dword [rbx + rcx*4]
0x100004188      add     rax, rbx
;-- switch
0x10000418b      jmp     rax       ; switch table (84 cases) at 0x1000048e8 ; the edges should be around here
```

Screenshot made using Cutter because it's easier to fit more info. But it was also confirmed using rizin from command line.

The part where switch edges get attached
Screenshot 2021-01-04 at 23 18 24

Thee part where edges should be attached
Screenshot 2021-01-04 at 23 22 51

This is just a guess, but maybe the basic block gets split after the switch edges are already created due to code jumping in the middle of it but the splitting logic leaves the edges attached to first part instead of second.

I will later try to make a test executable for this.

### PR DESCRIPTION
**Pull request by GustavoLCR**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

* Try to detect shift of first case value
* Detect when block where switch table is attached is split by analysis
* Fix attaching switch table to wrong block
* Add/fix tests

**Test plan**

Tests pass

**Closing issues**

Fixes #304

### CODE REVIEW

**Review by wargio — APPROVED** [review]

**Comment by wargio** [comment]
you need to rebase

**Comment by karliss** [comment]
Tested using the executable in #304 it now looks correct.

File: librz/analysis/jmptbl.c [hunk 1]
```diff
@@ -44,10 +44,26 @@ static void apply_switch(RzAnalysis *analysis, ut64 switch_addr, ut64 jmptbl_add
 // analyze a jmptablle inside a function // maybe rename to rz_analysis_fcn_jmptbl() ?
 RZ_API bool rz_analysis_jmptbl(RzAnalysis *analysis, RzAnalysisFunction *fcn, RzAnalysisBlock *block, ut64 jmpaddr, ut64 table, ut64 tablesize, ut64 default_addr) {
 	const int depth = 50;
-	return try_walkthrough_jmptbl(analysis, fcn, block, depth, jmpaddr, table, table, tablesize, tablesize, default_addr, false);
```
Comment (XVilka) [inline librz/analysis/jmptbl.c:47]: We need to simplify these functions in the future...

**Review by XVilka — APPROVED** [review]

File: test/db/analysis/x86_32 [hunk 2]
```diff
@@ -629,7 +629,7 @@ aaa
 f~?case
 EOF
 EXPECT=<<EOF
-90
+91
```
Comment (ret2libc) [inline test/db/analysis/x86_32:632]: I think this PR is changing this test in a more important way.
Old behaviour:
Screenshot 2021-01-23 at 16 06 05
New behaviour:
Screenshot 2021-01-23 at 16 06 16
Jump table:
Screenshot 2021-01-23 at 16 06 42

The basic block shown in the first two images is 0x10000192c + 4294965314 (modulo 32bit), which as you can see is the selected case of 1...14, etc., as shown in the old behaviour, not the new one.

**Review by ret2libc — CHANGES_REQUESTED** [review]

File: test/db/analysis/x86_32 (hunk 2 above)
Comment (karliss) [reply test/db/analysis/x86_32:632]: I am guessing that it's the result if implementing "shift of first case value" support. The question if in this case it's false posivite for shift logic or are the new results  more correct.

Comment (karliss) [reply test/db/analysis/x86_32:632]: image

The code subtracts 49 from the index before doing lookup in jump table. Seems likely that compiler removed first 49 unused entrie to reduce table size.

File: test/db/analysis/x86_64 [hunk 3]
```diff
@@ -3391,38 +3391,109 @@ f~case.
 EOF
 EXPECT=<<EOF
 0x140001023 1 switch.0x140001023
+0x140001f5b 1 switch.0x140001f5b
```
Comment (ret2libc) [inline test/db/analysis/x86_64:3394]: Maybe not for this PR, but this is really weird... There is a switch table here but there is no function defined. However at 0x140001e38 you can see `jbe   0x140001f40`, which goes just there. It's strange no basicblock was created there.

**Review by ret2libc — APPROVED** [review]

**Pull request merged**